pkg/plugin/aggregation/run.go

```diff
@@ -150,18 +150,46 @@ func Run(client kubernetes.Interface, plugins []plugin.Interface, cfg plugin.Agg
 	}()
 
 	// 4. Launch each plugin, to dispatch workers which submit the results back
-	for _, p := range plugins {
-		cert, err := auth.ClientKeyPair(p.GetName())
-		if err != nil {
-			return errors.Wrapf(err, "couldn't make certificate for plugin %v", p.GetName())
-		}
-		logrus.WithField("plugin", p.GetName()).Info("Running plugin")
-		if err = p.Run(client, cfg.AdvertiseAddress, cert); err != nil {
-			return errors.Wrapf(err, "error running plugin %v", p.GetName())
+	go func() {
+		pluginsRun := map[int]bool{}
+		for {
+			for index, p := range plugins {
```

Comment:
Previously we just synchronously started each plugin and now we just iterate over them checking if we can start them or not.
 - uses a map to check if we started each plugin (just using index of the plugin as a key)
 - didnt want to alter the slice of plugins since that is used elsewhere

Reply:
Where do we keep status of each plugin? Could you create a struct to store all the bits about the plugin (i.e. current status, start time, etc).